pkg/tsdb/azuremonitor/azcredentials/credentials.go

```diff
@@ -0,0 +1,30 @@
+package azcredentials
+
+const (
+	AzureAuthManagedIdentity = "msi"
+	AzureAuthClientSecret    = "clientsecret"
+)
+
+type AzureCredentials interface {
+	AzureAuthType() string
```

Comment:
maybe repeating `Azure` is not necessary? --> `AuthType() string`

Reply:
Go automatically matches structs to interfaces and there are other interfaces which implement `AuthType() string` :)
So I just made it unique enough to not clash with other interfaces.

My goal is to implement discriminated union type: `AzureCredentials` which is either `AzureManagedIdentityCredentials` or `AzureClientSecretCredentials`, basically [this](https://github.com/grafana/grafana/blob/b255f3db3f84d897f33deedb063e6ea5fd86eada/public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts#L74),

I saw on internet that people usually add a dummy method to interface like this:
```go
type MyUnionType interface {
     ...

     isMyUnionType() // dummy method to bind interface
}
```
Just for a purpose of binding the interface to implementation structs.

Example here:
https://medium.com/@haya14busa/sum-union-variant-type-in-go-and-static-check-tool-of-switch-case-handling-3bfc61618b1e

I decided to utilize the `AzureAuthType`, it has some practical use and also used to "glue" the interface to implementation instead of a dummy method.